background_worker/jobs/clear_metamorph.go

```diff
@@ -0,0 +1,42 @@
+package jobs
+
+import (
+	"fmt"
+	"github.com/jmoiron/sqlx"
+)
+
+func runDelete(table string, params ClearRecordsParams) error {
+	Log(INFO, "Connecting to database ...")
+	conn, err := sqlx.Open(params.Scheme(), params.String())
+	if err != nil {
+		Log(ERROR, fmt.Sprintf("unable to create connection %s", err))
+		return err
+	}
+	interval := fmt.Sprintf("%d days", params.RecordRetentionDays)
+
+	stmt, err := conn.Preparex(fmt.Sprintf("DELETE FROM %s WHERE inserted_at <= (CURRENT_DATE - $1::interval)", table))
+	if err != nil {
+		Log(ERROR, fmt.Sprintf("unable to prepare statement %s", err))
+		return err
+	}
+
+	res, err := stmt.Exec(interval)
+	if err != nil {
+		Log(ERROR, "unable to delete block rows")
+		return err
+	}
+	rows, _ := res.RowsAffected()
+	Log(INFO, fmt.Sprintf("Successfully deleted %d rows", rows))
+	return nil
+}
+
+func ClearMetamorph(params ClearRecordsParams) error {
+	if err := runDelete("blocks", params); err != nil {
```

Comment:
After adding the schema the from clause would be `metamorph.blocks` like here
https://github.com/bitcoin-sv/arc/blob/fbe226552c2e042b825bd50b90e0a4f3704ab837/metamorph/store/postgresql/postgres.go#L603

Reply:
I'm not sure it's a good idea to hardcode database name inside the query. Because if the name inside db connection can be different from the one inside the query.